game/src/ai.ts

```diff
@@ -0,0 +1,99 @@
+import { Game } from "./game";
+import { GameState, Move } from "./types";
+
+function calculateBestMove(
+  game: Game,
+  state: GameState,
+  depth: Number,
+  baseMove: Move,
+): Move {
+  let deck = state.decks[game.state.currentPlayer];
+
+  let bestMove = {
+    player: game.state.currentPlayer,
+    cardId: null,
+    placement: { row: 0, col: 0 },
+  };
+  let bestScore = -99;
+
+  if (baseMove) {
+    bestMove = baseMove;
+  }
+
+  // goes over all the deck cards
+  for (let deckCard of deck.cells) {
+    if (deckCard.cardId) {
+      let card = game.cards[deckCard.cardId];
+
+      if (typeof card.value === "number") {
+        // checks the entire board to see if it can be placed
+        for (let x = 0; x < game.numRows; x++) {
+          for (let y = 0; y < game.numCols; y++) {
+            // Does this current cell already occupy a card? If not, try placing (TODO: modifiers)
+            if (!state.board.cells[x * game.numRows + y].cardId) {
+              let newState = game.boardSetCard(
+                state.board,
+                { row: x, col: y },
+                card.id,
+              );
+
+              let tilesOwned = 0;
+              let valueOwned = 0;
+
+              for (let j = 0; j < newState.cells.length; j++) {
```

Comment:
Computing tilesOwned and valueOwned seems like a good candidate for its own function!